src/routes/synapse/room_events.rs

```diff
@@ -186,24 +188,53 @@ async fn process_room_event(
         friendships_repository: &repos.friendships,
         friendship_history_repository: &repos.friendship_history,
     };
+
+    // The only case where we don't create the friendship if it didn't exist
+    // If they are still no friends, it's unnecessary to create a friendship
+    if current_status == new_status {
+        return Ok(RoomEventResponse {
+            event_id: room_id.to_string(),
+        });
+    }
+
+    // Start transaction
+    let transaction = friendship_ports.db.start_transaction().await;
+    if transaction.is_err() {
```

Comment:
I think you could do something like this(instead of checking first the boolean and the unwrap) : 
```rust
if let Err(error) = transaction {
  //...
  log::error!(".... {err:?}")
  //...
}